impress-org/givewp
# Fix: fee recovery in the “donation summary” with wrong value

Resolves #6536

## Description



The donation summary is displaying the wrong fee recovery amount for currencies that uses a comma as a decimal separator. 

~~This PR solves the problem by adding extra param (with "true" as the default value) to the `format_amount()` function that allows us to set if the amount should be normalized or not because when it's called inside the `handleFees()` function the value passed through is already normalized.~~

~~So, when the value is already normalized, we need to set this extra param to "false" to prevent the double normalization process - because when it happens, a wrong value is generated.~~

This PR solves the problem by getting the formatted value directly from the DOM instead of retrieving it from data attributes and using the `format_amount()` function to normalize these values - in fact, this function is no longer useful and has been removed.

Another problem solved by this PR is that now the fee amount gets updated even when the `GiveWP > Settings > Fee Recovery > Fee Opt-In > Forced Opt-in` is enabled - this option removes the checkbox observed on the frontend and it also was preventing the fee amount update. 

Now the frontend no more observes the checkbox to handle the fee amount changes, instead that the `handleFees()` function is triggered every time that the `.give-final-total-amount` element gets updated. This way, some old codes responsible for updating the fee amount were removed from some files because they are no more necessary.

## Affects



## Visuals



Before fix: https://youtu.be/PcwG0vrvybU

After fix: https://youtu.be/jDG-YHV1ru8

## Testing Instructions



1. Install and active the Fee Recovery Add-on
2. Setup a multi-step or classic form template
3. Use a currency that uses a comma as a decimal separator like Euro (€)
4. Make a test donation and tick the "cover transaction cost" option
5. Before finishing the donation, the fee value in the “donation summary” should accurately reflect the fee recovery amount
6. Also make a second test donation with the `GiveWP > Settings > Fee Recovery > Fee Opt-In > Forced Opt-in` enabled

## Pre-review Checklist
